h2o-core/src/main/java/water/jdbc/SQLManager.java

```diff
@@ -38,207 +37,244 @@
-    } else {
-      vec = Vec.makeConN(numRow, num_chunks);
-    }
-    Log.info("Number of chunks for data retrieval: " + vec.nChunks() + ", number of rows:" + numRow);
-    //create frame
-    final Key<Frame> destination_key = Key.make((table + "_sql_to_hex").replaceAll("\\W", "_"));
-    final Job<Frame> j = new Job(destination_key, Frame.class.getName(), "Import SQL Table");
-
-    final String finalTable = table;
-    H2O.H2OCountedCompleter work = new H2O.H2OCountedCompleter() {
-      @Override
-      public void compute2() {
-        final ConnectionPoolProvider provider = new ConnectionPoolProvider(connection_url, username, password, vec.nChunks());
-        final Frame fr;
-
-        if (SqlFetchMode.DISTRIBUTED.equals(sqlFetchMode)) {
-          fr = new SqlTableToH2OFrame(finalTable, databaseType, columns, columnNames, numCol, j, provider)
-                  .doAll(columnH2OTypes, vec)
-                  .outputFrame(destination_key, columnNames, null);
-        } else {
-          fr = new SqlTableToH2OFrameStreaming(finalTable, databaseType, columns, columnNames, numCol, j, provider)
-                  .readTable(vec, columnH2OTypes, destination_key);
-        }
-        vec.remove();
+      Log.info("Number of chunks for data retrieval: " + vec.nChunks() + ", number of rows:" + numRow);
+      _j.setWork(vec.nChunks());
 
-        DKV.put(fr);
-        ParseDataset.logParseResults(fr);
-        if (finalTable.equals(SQLManager.TEMP_TABLE_NAME))
-          dropTempTable(connection_url, username, password);
-        tryComplete();
-      }
-    };
-    j.start(work, vec.nChunks());
+      // Finally read the data into an H2O Frame
+      _j.update(0L, "Importing data");
+      final String importTable = source_table;
+      final ConnectionPoolProvider provider = new ConnectionPoolProvider(_connection_url, _username, _password, vec.nChunks());
+      final Frame fr;
 
-    return j;
-  }
+      if (SqlFetchMode.DISTRIBUTED.equals(_fetch_mode)) {
+        fr = new SqlTableToH2OFrame(importTable, _database_type, _columns, columnNames, numCol, _j, provider)
+                .doAll(columnH2OTypes, vec)
+                .outputFrame(_destination_key, columnNames, null);
+      } else {
+        fr = new SqlTableToH2OFrameStreaming(importTable, _database_type, _columns, columnNames, numCol, _j, provider)
+                .readTable(vec, columnH2OTypes, _destination_key);
+      }
+      vec.remove();
 
-  private static Connection getConnectionSafe(String url, String username, String password) throws SQLException {
-    try {
-      return DriverManager.getConnection(url, username, password);
-    } catch (NoClassDefFoundError e) {
-      throw new RuntimeException("Failed to get database connection, probably due to using thin jdbc driver jar.", e);
+      DKV.put(fr);
+      ParseDataset.logParseResults(fr);
+      if (importTable.equals(SQLManager.TEMP_TABLE_NAME))
+        dropTempTable(_connection_url, _username, _password);
```

Comment:
I thought this drop was done in a finally clause at some point: as it is not a true temporary table, it's important to have this always executed otherwise it prevents the user from running imports later, even if s/he fixed the driver issue for example.